core/src/main/java/org/apache/iceberg/rest/RESTTableOperations.java

```diff
@@ -155,20 +157,86 @@ public void commit(TableMetadata base, TableMetadata metadata) {
     // the error handler will throw necessary exceptions like CommitFailedException and
     // UnknownCommitStateException
     // TODO: ensure that the HTTP client lib passes HTTP client errors to the error handler
-    LoadTableResponse response =
-        client.post(path, request, LoadTableResponse.class, headers, errorHandler);
+    LoadTableResponse response;
+    try {
+      response = client.post(path, request, LoadTableResponse.class, headers, errorHandler);
+    } catch (CommitStateUnknownException e) {
+      // Lightweight reconciliation for snapshot-add-only updates on transient unknown commit state
+      if (updateType == UpdateType.SIMPLE && reconcileOnSimpleUpdate(updates, e)) {
+        return;
+      }
+
+      throw e;
+    }
 
     // all future commits should be simple commits
     this.updateType = UpdateType.SIMPLE;
 
     updateCurrentMetadata(response);
```

Comment:
This will be skipped in case of reconciling. So that `this.current` may not be updated. Is that a concern?